raiden/transfer/mediated_transfer/initiator_manager.py

```diff
@@ -214,38 +288,41 @@ def handle_transferrefundcancelroute(
     )
 
     events = list()
-    if is_valid_lock and is_valid_refund:
-        is_valid, channel_events, _ = channel.handle_receive_refundtransfercancelroute(
-            channel_state,
-            refund_transfer,
-        )
+    if not is_valid_lock or not is_valid_refund:
+        return TransitionResult(payment_state, list())
 
-        events.extend(channel_events)
-
-        if is_valid:
-            old_description = payment_state.initiator.transfer_description
-            transfer_description = TransferDescriptionWithSecretState(
-                old_description.payment_network_identifier,
-                old_description.payment_identifier,
-                old_description.amount,
-                old_description.token_network_identifier,
-                old_description.initiator,
-                old_description.target,
-                state_change.secret,
-            )
-            payment_state.initiator.transfer_description = transfer_description
-
-            sub_iteration = handle_cancelroute(
-                payment_state,
-                state_change,
-                channelidentifiers_to_channels,
-                pseudo_random_generator,
-                block_number,
-            )
+    is_valid, channel_events, _ = channel.handle_receive_refundtransfercancelroute(
+        channel_state,
+        refund_transfer,
+    )
+
+    events.extend(channel_events)
 
-            events.extend(sub_iteration.events)
-            if sub_iteration.new_state is None:
-                payment_state = None
+    if not is_valid:
+        return TransitionResult(payment_state, list())
+
```

Comment:
we need to call `cancel_route` here, otherwise the transfer will not be moved from the mapping to the cancelled list

Reply:
This is already done in `try_new_route`

Reply:
Same error as before, I though the transfer was removed from the mapping when cancelled.